dbms/src/Interpreters/ExpressionAnalyzer.h

```diff
@@ -219,6 +219,11 @@ class SelectQueryExpressionAnalyzer : public ExpressionAnalyzer
       */
     void tryMakeSetForIndexFromSubquery(const ASTPtr & subquery_or_table_name);
 
+    /**
+    * Checks if subquery is not a plain StorageSet.
+    */
+    bool checkIfPossibleToMakeSetForIndexFromSubquery(const ASTPtr & subquery_of_table_name);
```

Comment:
There's a huge unreadable fuctions name that even needs a comment to be undestood. It's better to rename starting with ```support``` or ```allow``` instead of ```checkIfPossibleToMake```